File: OSS-Encrypt-Go/OSS-Encrypt.go
// Test OSS server side encryption feature.
package main

import (
	"fmt"
	"os"
	"strings"

	"github.com/aliyun/aliyun-oss-go-sdk/oss"
)

const (
	// TokyoOSSEndPoint is the Tokyo region endpoint
	TokyoOSSEndPoint = "https://oss-ap-northeast-1.aliyuncs.com"
	// AccessKeyID should should not be included in a source code like this
	AccessKeyID = "<Your AccessKeyId>"
	// AccessKeySecret should not be included in a source code like this
	AccessKeySecret = "<Your AccessKeySecret>"
)

func handleError(err error) {
	fmt.Println("Error:", err)
	os.Exit(-1)
}
func main() {
	bucketName := "<Your Bucket>"

	// Initialize the client
	client, err := oss.New(TokyoOSSEndPoint, AccessKeyID, AccessKeySecret)
	if err != nil {
		handleError(err)
	}

	// Get the bucket
	bucket, err := client.Bucket(bucketName)
	if err != nil {
		handleError(err)
	}

	// Upload a file to OSS with server side encryption enabled. We use OSS and KMS integration feature to automatically encrypt file on upload.
	// For every region, if server side encryption is used for the first time with KMS, OSS will automatically create am encryption key on KMS first. KMS must be activated before this operation.
	err = bucket.PutObject("test.txt", strings.NewReader("Never Give Up - Jack Ma."), oss.ServerSideEncryption("KMS"))
	if err != nil {
		handleError(err)
	}

	// We can then get the object and print it to verify that it has been encrypted using a key generated on KMS
	object, err := bucket.GetObject("test.txt")
	if err != nil {
		handleError(err)
	}
	fmt.Print(object)
}

File: OSS-Encrypt-Python/OSS-Encrypt.py
# -*- coding: utf-8 -*-

import os
import shutil

import oss2


access_key_id = os.getenv('OSS_TEST_ACCESS_KEY_ID', '<Your AccessKeyId>')
access_key_secret = os.getenv('OSS_TEST_ACCESS_KEY_SECRET', '<Your AccessKeySecret>')
bucket_name = os.getenv('OSS_TEST_BUCKET', '<Your Bucket>')
endpoint = os.getenv('OSS_TEST_ENDPOINT', 'http://oss-ap-northeast-1.aliyuncs.com')


# Bucketを作成する
bucket = oss2.Bucket(oss2.Auth(access_key_id, access_key_secret), endpoint, bucket_name)


# Objectを作成する。暗号化するためにKMSを指定する。
bucket.put_object('motto.txt', 'Never give up. - Jack Ma', headers={"x-oss-server-side-encryption" : "KMS"})